src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/TextFilter.php

```diff
@@ -139,4 +164,18 @@ protected function checkValue(AttributeInterface $attribute, $value)
             throw InvalidPropertyTypeException::stringExpected($attribute->getCode(), static::class, $value);
         }
     }
+
+    /**
+     * Escapes value
+     *
+     * @param string $value
+     *
+     * @return string
+     */
+    protected function escapeValue($value)
+    {
+        $regex = '#[-+=|! &(){}\[\]^"~*<>?:/\\\]#';
```

Comment:
I don't understand why you have this in the 2 filters
I thought the main purpose of splitting the StringFilter into Text and TextArea filters was because we didn't have to execute the same queries